sshd-core/src/main/java/org/apache/sshd/client/config/hosts/DefaultConfigFileHostEntryResolver.java

```diff
@@ -75,12 +90,105 @@ public final boolean isStrict() {
                 throw new IOException("String permission violation (" + violation + ") for " + path);
             }
+
+            String ownerViolation = validateStrictConfigFileOwner(path);
+            if (ownerViolation != null) {
+                throw new IOException("String owner violation (" + ownerViolation + ") for " + path);
+            }
         }
 
         return super.reloadHostConfigEntries(path, host, port, username);
     }
+
+    /**
+     * <P>Checks if a path has strict permissions</P>
+     * <UL>
+     *
+     * <LI><P>
+     * (For {@code Unix}) The path may not have group or others write permissions
+     * </P></LI>
+     *
+     * </UL>
+     *
+     * @param path    The {@link Path} to be checked - ignored if {@code null}
+     *                or does not exist
+     * @param options The {@link LinkOption}s to use to query the file's permissions
+     * @return The violated {@link PosixFilePermission} - {@code null} if
+     * no violations detected
+     * @throws IOException If failed to retrieve the permissions
+     * @see #STRICTLY_PROHIBITED_FILE_PERMISSION
+     */
+    public static PosixFilePermission validateStrictConfigFilePermissions(Path path, LinkOption... options) throws IOException {
+        if ((path == null) || (!Files.exists(path, options))) {
+            return null;
+        }
+
+        Collection<PosixFilePermission> perms = IoUtils.getPermissions(path, options);
+        if (GenericUtils.isEmpty(perms)) {
+            return null;
+        }
+
+        if (OsUtils.isUNIX()) {
+            PosixFilePermission p = IoUtils.validateExcludedPermissions(perms, STRICTLY_PROHIBITED_FILE_PERMISSION);
+            if (p != null) {
+                return p;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * <P>Checks if a path has strict ownership</P>
+     * <UL>
+     *
+     * <LI><P>
+     * The path must be owned by current user.
+     * </P></LI>
+     *
+     * <LI><P>
+     * (For {@code Unix}) The path may be owned by root.
+     * </P></LI>
+     *
+     * </UL>
+     *
+     * @param path    The {@link Path} to be checked - ignored if {@code null}
+     *                or does not exist
+     * @param options The {@link LinkOption}s to use to query the file's permissions
+     * @return The violated owner - {@code null} if
+     * no violations detected
+     * @throws IOException If failed to retrieve the ownership
+     */
+    public static String validateStrictConfigFileOwner(Path path, LinkOption... options) throws IOException {
+        if ((path == null) || (!Files.exists(path, options))) {
+            return null;
+        }
+
+        String current = IoUtils.getCurrentUser();
+        String owner = IoUtils.getFileOwner(path, options);
+
+        if (current == null &&  owner == null) {
+            // we cannot detect permissions
+            return null;
+        }
+
+        Set<String> expected = new HashSet<>();
```

Comment:
Use:
```java
Set<String> expected = new TreeSet<>(String.CASE_INSENSITITVE_ORDER);
```
Personally, I think that anyone using "ROOT" and "root" as distinct users (for Linux) or "administrator" and "ADMINISTRATOR" (Windows) deserve all they get. However, perhaps this may be better:
```java
Set<String> expected = OsUtils.isWin32() ? new TreeSet<>(String.CASE_INSENSITITVE_ORDER) : new HashSet<>();
```